dora/core/common/src/main/java/alluxio/membership/AlluxioEtcdClient.java

```diff
@@ -127,36 +126,41 @@ public static AlluxioEtcdClient getInstance(AlluxioConfiguration conf) {
     return sAlluxioEtcdClient;
   }
 
-  /**
-   * Create jetcd grpc client no forcing.
-   */
-  public void connect() {
-    connect(false);
+  @FunctionalInterface
+  protected interface EtcdUtilCallable<V> {
+    /**
+     * The task where logics to communicate with etcd happens.
+     *
+     * @return Etcd gRPC call result
+     * @throws Exception
+     */
+    V call() throws Exception;
   }
 
-  /**
-   * Create jetcd grpc client and force(or not) connection.
-   * @param force
-   */
-  public void connect(boolean force) {
-    if (mConnected.get() && !force) {
-      return;
-    }
-    mConnected.set(false);
-    // create client using endpoints
-    Client client = Client.builder().endpoints(mEndpoints)
-        .build();
-    if (mConnected.compareAndSet(false, true)) {
-      mClient = client;
+  private <V> V retryInternal(String description, RetryPolicy retryPolicy,
+                              EtcdUtilCallable<V> etcdCallable) throws IOException {
+    Exception ex = null;
+    // TODO(lucy) Currently retry on all sorts of exception and report the last exception,
+    // As jetcd exception often hides underneath CompletableFuture.get(), find
+    // more info later on distinguishing different possible exceptions.
+    while (retryPolicy.attempt()) {
+      try {
+        return etcdCallable.call();
+      } catch (Exception e) {
```

Comment:
retry for whatever reason?

What if it is the fatal error? eg, path not exist?

Reply:
oh that's not fatal at all, etcd will not throw exception bcos certain path doesn't exist.
Currently EtcdException( the jetcd exception) is mostly for connection related, usually manifest itself in callback to those StreamObservers,  I haven't captured in inline call, we get more experience playing with jetcd and handle later